lib/liquid/standardfilters.rb

```diff
@@ -65,9 +65,10 @@ def truncate(input, length = 50, truncate_string = "...".freeze)
       return if input.nil?
       input_str = input.to_s
       length = Utils.to_integer(length)
-      l = length - truncate_string.length
+      truncate_string_str = truncate_string.to_s
```

Comment:
I'm not sure if introducing a new variable is actually necessary. This simply follows what was done for `input_str`. I personally don't have a problem with converting and reassigning the value of a variable even if it means a type change.